tests/utils/test_archive.py

```diff
@@ -53,43 +53,36 @@ def _parse_archive_name(name):
         incarnation_no_ext = os.path.splitext(incarnation_ext)[0]
         return timestamp_str, incarnation_no_ext
 
-    def test_archive01(self):
-        """
-        StateArchiver should archive all history directories by
-
-          1. Creating a .zip of a timestamped directory's files
-          2. Saving the .zip to /var/lib/waagent/history/
-          2. Deleting the timestamped directory
-        """
-        temp_files = [
+    def test_archive_should_zip_all_but_the_latest_goal_state_in_the_history_folder(self):
+        test_files = [
             'GoalState.xml',
             'Prod.manifest.xml',
             'Prod.agentsManifest',
             'Microsoft.Azure.Extensions.CustomScript.xml'
         ]
 
-        # this directory matches the pattern that StateArchiver.archive() searches for
-        temp_directory = os.path.join(self.history_dir, datetime.utcnow().isoformat() + "_incarnation_0")
-
-        for current_file in temp_files:
-            self._write_file(os.path.join(temp_directory, current_file))
+        # these directories match the pattern that StateArchiver.archive() searches for
+        test_directories = []
+        for i in range(0, 3):
+            timestamp = (datetime.utcnow() + timedelta(minutes=i)).isoformat()
+            directory = os.path.join(self.history_dir, "{0}_incarnation_{1}".format(timestamp, i))
+            for current_file in test_files:
+                self._write_file(os.path.join(directory, current_file))
+            test_directories.append(directory)
 
         test_subject = StateArchiver(self.tmp_dir)
         test_subject.archive()
 
-        timestamp_zips = os.listdir(self.history_dir)
-        self.assertEqual(1, len(timestamp_zips))
+        for directory in test_directories[0:1]:
```

Comment:
This index only checks one directory, so could change to test second one too?

Reply:
this checks items 0 & 1, item 2 is checked at line 85

Reply:
fixed